controllers/componentdetectionquery_controller.go

```diff
@@ -109,28 +111,83 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
 		devfilesURLMap := make(map[string]string)
 		dockerfileContextMap := make(map[string]string)
 
+		// check if the project is multi-component or single-component
+		log.Info(fmt.Sprintf("Look for devfile or dockerfile at the root dir... %v", req.NamespacedName))
+		isMultiComponent := false
+		isDockerfilePresent := false
+		isDevfilePresent := false
+
+		var gitURL string
+		if gitToken == "" {
+			gitURL, err = util.ConvertGitHubURL(source.URL, source.Revision)
+			if err != nil {
+				log.Error(err, fmt.Sprintf("Unable to convert Github URL to raw format, exiting reconcile loop %v", req.NamespacedName))
+				r.SetCompleteConditionAndUpdateCR(ctx, &componentDetectionQuery, err)
+				return ctrl.Result{}, nil
+			}
+
+			devfileBytes, dockerfileBytes = devfile.DownloadDevfileAndDockerfile(gitURL)
+		} else {
+			// Use SPI to retrieve the devfile from the private repository
+			// TODO - maysunfaisal also search for Dockerfile
+			devfileBytes, err = spi.DownloadDevfileUsingSPI(r.SPIClient, ctx, componentDetectionQuery.Namespace, source.URL, "main", "")
+			if err != nil {
+				log.Error(err, fmt.Sprintf("Unable to curl for any known devfile locations from %s %v", source.URL, req.NamespacedName))
+			}
+		}
+
+		isDevfilePresent = len(devfileBytes) != 0
+		isDockerfilePresent = len(dockerfileBytes) != 0
+
+		if isDevfilePresent {
+			log.Info(fmt.Sprintf("Found a devfile, devfile to be analyzed to see if a Dockerfile is referenced %v", req.NamespacedName))
+			devfilesMap["./"] = devfileBytes
+		} else if isDockerfilePresent {
+			log.Info(fmt.Sprintf("Determined that this is a Dockerfile only component  %v", req.NamespacedName))
+			dockerfileContextMap["./"] = "./Dockerfile"
+		}
+
+		// Clone the repo if no dockerfile present
+		if !isDockerfilePresent {
+			log.Info(fmt.Sprintf("Unable to find devfile or Dockerfile under root directory, run Alizer to detect components... %v", req.NamespacedName))
```

Comment:
I am confused on why we check for Alizer components only when no dockerfile is not found

Should it be if `!isDevfilePresent && !isDockerfilePresent {` ?

Because if a devfile is found in the root dir, dont we just take it a single component with devfile in the root dir and skip all this?

Reply:
This is the old log: https://github.com/redhat-appstudio/application-service/blob/9c6a8ee09b046e66e90957dab8be574925ee3bb2/controllers/componentdetectionquery_controller.go#L177
```
if (!isDevfilePresent && !isDockerfilePresent) || (isDevfilePresent && !isDockerfilePresent) 
```
is basically `!isDockerfilePresent`.

you are right, if a devfile or dockerfile is found in root dir, we will take it as a single component. 
but if only dockerfile is found, need to run `analyzePath` to find a match devfile (check previous link for the logic).
 
so in this case, I simplify the code to clone at the beginning when no dockerfile present.. and can directly analyzePath [here](https://github.com/redhat-appstudio/application-service/blob/4854e6b9a7705ffc217b61a6bf2e3a1efa66fa65/controllers/componentdetectionquery_controller.go#L201).

to your concern, I have updated the code to only run Alizer if `!isDevfilePresent`.